### PR TITLE
[Sil] [ProductBundle] Wip product variants

### PR DESCRIPTION
**Pull request by PapsOu**

### CODE REVIEW

File: src/PlatformBundle/Entity/SilStockBundle/StockItem.php [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+     */
+    protected $outputStrategy;
+
+    /**
+     * @param CodeInterface  $code
+     * @param string         $name
+     * @param Uom            $uom
+     * @param OutputStrategy $strategy
+     */
+    public static function creatDefault(CodeInterface $code, string $name, Uom $uom, OutputStrategy $strategy)
```
Comment (RomainSanchez) [inline src/PlatformBundle/Entity/SilStockBundle/StockItem.php]: 'creatDefault'

File: src/Sil/Bundle/ProductBundle/Form/Transformer/ProductToIdTransformer.php [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    /**
+     * @var ProductRepositoryInterface
+     */
+    protected $productRepository;
+
+    public function transform($value): ?string
+    {
+        if ($value instanceof ProductInterface) {
+            return $value->getId();
+        } else {
```
Comment (RomainSanchez) [inline src/Sil/Bundle/ProductBundle/Form/Transformer/ProductToIdTransformer.php]: No need for else statement there

File: src/Sil/Bundle/ProductBundle/Generator/ProductVariantGenerator.php [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+                foreach ($option as $optionType => $optionValue) {
+                    $variantName .= ' ' . $optionValue->getValue();
+
+                    $variant->addOption($optionValue);
+                }
+
+                $variant->setName($variantName);
+            }
+        } else {
+            // Add a default product variant if not option has been set
```
Comment (RomainSanchez) [inline src/Sil/Bundle/ProductBundle/Generator/ProductVariantGenerator.php]: 'no' option has been set

File: src/Sil/Bundle/ProductBundle/Repository/ProductVariantRepository.php [hunk 4]
```diff
@@ -42,4 +45,20 @@ public function getInactiveVariantsForProduct(ProductInterface $product): Collec
     {
         return $this->findBy(['product' => $product, 'active' => false]);
     }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function findByCode(CodeInterface $code): ?ResourceInterface
```
Comment (RomainSanchez) [inline src/Sil/Bundle/ProductBundle/Repository/ProductVariantRepository.php]: Shouldn't the name be FindOneByCode ?

File: src/Sil/Bundle/ProductBundle/Resources/views/ProductVariant/variants.html.twig [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+
+                </div>
+
+            </div>
+
+        {% endfor %}
+
+    </div>
+
+    {{ form_rest(form)}}
```
Comment (RomainSanchez) [inline src/Sil/Bundle/ProductBundle/Resources/views/ProductVariant/variants.html.twig]: form_end already calls form_rest

File: src/Sil/Bundle/ProductBundle/Repository/ProductVariantRepository.php (hunk 4 above)
Comment (PapsOu) [reply src/Sil/Bundle/ProductBundle/Repository/ProductVariantRepository.php]: Nice catch !

**Pull request merged**